app/src/main/java/rick_and_morty/ui/episodes/EpisodesViewModel.kt

```diff
@@ -1,18 +1,25 @@
 package rick_and_morty.ui.episodes
 
+import android.util.Log
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import io.realm.Realm
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import rick_and_morty.data.model.episodes.EpisodesResultDto
+import rick_and_morty.data.model.episodes.realm.RealmEpisodes
 import rick_and_morty.data.repository.EpisodesRepository
+import rick_and_morty.ui.episodes.EpisodesMapper.toEpisodesResultDto
+import rick_and_morty.ui.episodes.EpisodesMapper.toRealmEpisode
 import javax.inject.Inject
 
 @HiltViewModel
 class EpisodesViewModel @Inject constructor(
     private val episodesRepository: EpisodesRepository,
+    private val realm: Realm, // Inject an instance of Realm
```

Comment:
Since the comment is stating what the code is obviously doing it can be omitted.

More important: We should access the database via the repository and not directly in the view model. This way we would separate the layers of the architecture dealing with domain and data related logic.

https://developer.android.com/topic/architecture